dsl/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+extern crate quote;
+extern crate litrs;
+extern crate proc_macro2;
+
+use quote::quote;
+
+use proc_macro2::TokenTree;
+use proc_macro2::TokenStream;
+use proc_macro2::TokenTree::Punct;
+use proc_macro2::TokenTree::Ident;
+use proc_macro2::token_stream::IntoIter;
+use std::iter::Peekable;
+
+use std::convert::TryFrom;
+use std::str::FromStr;
+use litrs::Literal;
+
+#[derive(PartialEq)]
+enum ExprType {
+    Nil, Int, Bool
```

Comment:
Familiar name. 

I suggest you have a look at [non_exhaustive](https://doc.rust-lang.org/reference/attributes/type_system.html) and see if it fits the role of `Nil`.

Reply:
I think it would work just fine, i'll make the relevant changes